### PR TITLE
Adding range function for python

### PR DESCRIPTION
**Pull request by riknoll**
Fixes https://github.com/microsoft/pxt-microbit/issues/3104

This adds an implementation for `range()`. It does not change anything about for-loops, they still get mapped the same way and they still can only take one argument in the call to `range()`.

Requires https://github.com/microsoft/pxt-microbit/pull/3109

### CODE REVIEW

File: pxtcompiler/emitter/service.ts [hunk 1]
```diff
@@ -31,7 +31,8 @@ namespace ts.pxtc {
         "!!": { n: "bool", t: ts.SyntaxKind.BooleanKeyword },
         "Array.indexOf": { n: "Array.index", t: ts.SyntaxKind.Unknown },
         "Array.push": { n: "Array.append", t: ts.SyntaxKind.Unknown },
-        "parseInt": { n: "int", t: ts.SyntaxKind.NumberKeyword, snippet: 'int("0")' }
+        "parseInt": { n: "int", t: ts.SyntaxKind.NumberKeyword, snippet: 'int("0")' },
+        "_py.range": { n: "range", t: ts.SyntaxKind.Unknown, snippet: 'range("0")' }
```
Comment (pelikhan) [inline pxtcompiler/emitter/service.ts:35]: ``range(4)`` ?

**Comment by riknoll** [comment]
oh.... yes....

**Comment by riknoll** [comment]
https://github.com/microsoft/pxt/pull/7065

**Comment by pelikhan** [comment]
range wasn’t going down without a fight

________________________________

**Pull request merged**